controllers/infrastructure/byomachine_controller.go

```diff
@@ -537,3 +619,64 @@ func (r *ByoMachineReconciler) markHostForCleanup(ctx context.Context, machineSc
 	// Issue the patch for byohost
 	return helper.Patch(ctx, machineScope.ByoHost)
 }
+
+func (r *ByoMachineReconciler) getInstallerConfig(ctx context.Context, byoMachine *infrav1.ByoMachine) (*unstructured.Unstructured, error) {
+	installerConfig := &unstructured.Unstructured{}
+	gvk := byoMachine.Spec.InstallerRef.GroupVersionKind()
+	gvk.Kind = strings.Replace(gvk.Kind, "Template", "", -1)
+	installerConfig.SetGroupVersionKind(gvk)
+	installerName := client.ObjectKey{
+		Namespace: byoMachine.Namespace,
+		Name:      byoMachine.Name,
+	}
+	err := r.Client.Get(ctx, installerName, installerConfig)
+	if err != nil {
+		return nil, err
+	}
+	return installerConfig, nil
+}
+
+func (r *ByoMachineReconciler) createInstallationConfig(ctx context.Context, machineScope *byoMachineScope) error {
+	logger := log.FromContext(ctx).WithValues("cluster", machineScope.Cluster.Name)
+	var (
+		installerConfig *unstructured.Unstructured
+		err             error
+	)
+	_, err = r.getInstallerConfig(ctx, machineScope.ByoMachine)
+	if err != nil && apierrors.IsNotFound(err) {
+		template := &unstructured.Unstructured{}
+		template.SetGroupVersionKind(machineScope.ByoMachine.Spec.InstallerRef.GroupVersionKind())
+		installerTemplateName := client.ObjectKey{
+			Namespace: machineScope.ByoMachine.Spec.InstallerRef.Namespace,
+			Name:      machineScope.ByoMachine.Spec.InstallerRef.Name,
+		}
+		if err = r.Client.Get(ctx, installerTemplateName, template); err != nil {
+			logger.Error(err, "failed to get installer config template")
+			return err
+		}
+		installerAnnotations := map[string]string{
+			infrav1.K8sVersionAnnotation: strings.Split(*machineScope.Machine.Spec.Version, "+")[0],
+		}
+		installerConfig, err = external.GenerateTemplate(&external.GenerateTemplateInput{
+			Template:    template,
+			TemplateRef: machineScope.ByoMachine.Spec.InstallerRef,
+			Namespace:   machineScope.Cluster.Namespace,
```

Comment:
nitpick: just to most of the info from `ByoMachine` if available.
```suggestion
			Namespace:   machineScope.ByoMachine.Namespace,
```